src/Storages/transformQueryForExternalDatabase.cpp

```diff
@@ -154,20 +156,30 @@ bool isCompatible(IAST & node)
             && (function->arguments->children.size() != 2 || function->arguments->children[1]->as<ASTTableIdentifier>()))
             return false;
 
-        for (const auto & expr : function->arguments->children)
-            if (!isCompatible(*expr))
+        for (auto & expr : function->arguments->children)
+            if (!isCompatible(expr))
                 return false;
 
         return true;
     }
 
-    if (const auto * literal = node.as<ASTLiteral>())
+    if (const auto * literal = node->as<ASTLiteral>())
     {
+        if (literal->value.getType() == Field::Types::Tuple)
+        {
+            /// Represent a tuple with zero or one elements as (x) instead of tuple(x).
+            auto tuple_value = literal->value.safeGet<Tuple>();
+            if (tuple_value.size() == 1)
+            {
+                node = makeASTFunction("", std::make_shared<ASTLiteral>(tuple_value[0]));
```

Comment:
In old interpreter it's covered by this code https://github.com/ClickHouse/ClickHouse/blob/04822a63e1e5f0a236c3a247340161b3b74647d4/src/Storages/transformQueryForExternalDatabase.cpp#L146-L149

In new we have literal instead of function here